crates/containerd-shim-wasm/src/sandbox/shim/otel.rs

```diff
@@ -83,34 +106,33 @@ impl Config {
         Ok(serde_json::to_string(&injector)?)
     }
 
+    /// Sets the trace context from a JSON string.
     pub fn set_trace_context(trace_context: &str) -> anyhow::Result<()> {
         let extractor: HashMap<String, String> = serde_json::from_str(trace_context)?;
         let context = global::get_text_map_propagator(|propagator| propagator.extract(&extractor));
         Span::current().set_parent(context);
         Ok(())
     }
 
-    fn init_tracer_http_protobuf(&self) -> SpanExporterBuilder {
+    fn init_tracer_http(&self) -> SpanExporterBuilder {
         opentelemetry_otlp::new_exporter()
             .http()
-            .with_endpoint(&self.otel_endpoint)
+            .with_endpoint(&self.traces_endpoint)
             .into()
     }
 
     fn init_tracer_grpc(&self) -> SpanExporterBuilder {
         opentelemetry_otlp::new_exporter()
             .tonic()
-            .with_endpoint(&self.otel_endpoint)
+            .with_endpoint(&self.traces_endpoint)
             .into()
     }
 
     fn init_tracer(&self) -> Result<opentelemetry_sdk::trace::Tracer, TraceError> {
-        let exporter = match self.otel_protocol.as_str() {
-            OTEL_EXPORTER_OTLP_PROTOCOL_HTTP_PROTOBUF => self.init_tracer_http_protobuf(),
-            OTEL_EXPORTER_OTLP_PROTOCOL_GRPC => self.init_tracer_grpc(),
-            _ => Err(TraceError::from(
-                "Invalid OTEL_EXPORTER_OTLP_PROTOCOL value",
-            ))?,
+        let exporter = match self.traces_protocol {
+            Protocol::HttpBinary => self.init_tracer_http(),
+            Protocol::HttpJson => self.init_tracer_http(),
+            Protocol::Grpc => self.init_tracer_grpc(),
```

Comment:
**praise:** Great! Now we get a compile error if upstream adds a new protocol, and we can fix it :-)